### DISCUSSION TITLE
`sphinx_theme_builder._internal.cli` calls `sys.exit()` on module import, breaking pydoc, module introspection

### DISCUSSION THREAD

**Discussion by ferdnyc** [pradyunsg/sphinx-theme-builder#40] [Potential Issue]
If a project dependency is missing, the `__init__.py` for `sphinx_theme_builder._internal.cli` will call `sys.exit(1)` _on **import**_, making it impossible to examine the module via normal Python introspection tools like `importlib.import_module()` or `pydoc`.

Worse, a user who attempts to run a documentation keyword search from the command line using `pydoc -k <keyword>` will find their scan abruptly terminates as soon as pydoc reaches the module.

Even just typing `import sphinx_theme_builder._internal.cli` in a REPL will instantly drop the user out of their Python session, destroying any and all work they'd done with it up to that point:

```python
$ python3
Python 3.11.2 (main, Feb  8 2023, 00:00:00) [GCC 12.2.1 20221121 (Red Hat 12.2.1-4)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sphinx_theme_builder._internal.cli
error: missing-command-line-dependencies

× Could not import a package that is required for the `stb` command line.
╰─> No module named 'sphinx_autobuild'

note: The optional cli dependencies of this package are missing.
hint: During installation, make sure to include the `[cli]`. For example:
      pip install "sphinx-theme-builder[cli]"
link: 
https://sphinx-theme-builder.rtfd.io/errors/#missing-command-line-dependencies

$
```

`sys.exit()` should only ever be called either inside a function (which isn't executed on import), or when inside an `if __name__ == '__main__':` conditional block. It's **never** OK to take down the entire interpreter with an `exit()` call at the module level, where it can be executed on import. (_Especially_ for "optional" dependencies!) Following that rule is just part of being a polite and respectful member of the package ecosystem.

**Comment by pradyunsg** [comment]
Why are you importing an `_internal` submodule directly?

**Reply by ferdnyc** [reply]
As I said, one reason is when doing a command-line `pydoc -k` search.

**Reply by pradyunsg** [reply]
Adding a cross-reference for posterity: https://discuss.python.org/t/guidance-on-importability-of-installed-modules/25422

**Reply by pradyunsg** [reply]
> Specifying a -k flag before the argument will search the synopsis lines of all available modules for the keyword given as the argument, again in a manner similar to the Unix man command. The synopsis line of a module is the first line of its documentation string.

Fair enough -- I've elaborated on my concerns over on the discuss.python.org thread. :)
